Library/Homebrew/test/test_integration_cmds.rb

```diff
@@ -13,8 +13,28 @@ def setup
   end
 
   def teardown
-    (HOMEBREW_PREFIX/"bin").rmtree
-    @formula_files.each(&:unlink)
+    coretap = CoreTap.new
+    files_to_delete = []
+    files_to_delete += HOMEBREW_CELLAR.children
+    files_to_delete += HOMEBREW_CACHE.children
+    files_to_delete += HOMEBREW_LOCK_DIR.children
+    files_to_delete += HOMEBREW_LOGS.children
+    files_to_delete += [
+      HOMEBREW_PREFIX/"bin",
+      HOMEBREW_PREFIX/"share",
+      HOMEBREW_PREFIX/"opt",
+      HOMEBREW_LIBRARY/"LinkedKegs",
+      HOMEBREW_LIBRARY/"Taps/caskroom",
+      HOMEBREW_LIBRARY/"Taps/homebrew/homebrew-bundle",
+      HOMEBREW_LIBRARY/"Taps/homebrew/homebrew-foo",
+      HOMEBREW_LIBRARY/"Taps/homebrew/homebrew-services",
+      HOMEBREW_LIBRARY/"Taps/homebrew/homebrew-shallow",
+      HOMEBREW_REPOSITORY/".git",
+      coretap.formula_dir.children,
+      coretap.alias_dir,
+      coretap.path/".git",
```

Comment:
Can we combine all the `coretap.<thing>` cases and just use `coretap.path.children`? Or do we explicitly need to preserve something in this directory?